tasks/coverage/src/suite.rs

```diff
@@ -285,6 +285,12 @@ pub trait Case: Sized + Sync + Send + UnwindSafe {
             .with_module_record_builder(true)
             .with_check_syntax_error(true)
             .build(program);
+        if !source_type.is_typescript_definition() && !source_type.is_typescript() {
```

Comment:
This logic doesn't belong here in the suite.

We should add a trait method such as `check_semantic` to this suite, and then override it in `test262.rs`.